File: trackflow-backend/src/main/java/com/trackflow/module/report/service/ReportService.java
package com.trackflow.module.report.service;

import com.trackflow.module.report.dto.ReportRequest;
import com.trackflow.module.report.dto.ReportResponse;
import org.springframework.core.io.Resource;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface ReportService {
    ReportResponse generateReport(ReportRequest request);
    Page<ReportResponse> getMyReports(Pageable pageable);
    Resource downloadReport(UUID reportId);
}

File: trackflow-frontend/src/services/reportService.js
import api from './api'

export const reportService = {
  generateReport: (data) => api.post('/reports', data),
  getMyReports: (params) => api.get('/reports', { params }),
  downloadReport: (id) => api.get(`/reports/${id}/download`, {
    responseType: 'blob'
  }),
}